Cannot npm install posix@4.1.0 on Centos6.5

Failing on npm install on centos6.5

Error: Failed at the posix@4.1.0 install script 'node-gyp rebuild'


> posix@4.1.0 install /home/vagrant/blah/node_modules/posix
> node-gyp rebuild

make: Entering directory `/home/vagrant/blah/node_modules/posix/build'
  CXX(target) Release/obj.target/posix/src/posix.o
../src/posix.cc: In function ‘Nan::NAN_METHOD_RETURN_TYPE node_update_swap_constants(Nan::NAN_METHOD_ARGS_TYPE)’:
../src/posix.cc:653:78: error: ‘SWAP_FLAG_DISCARD’ was not declared in this scope
     obj->Set(Nan::New<String>("discard").ToLocalChecked(), Nan::New<Integer>(SWAP_FLAG_DISCARD));
                                                                              ^
make: *** [Release/obj.target/posix/src/posix.o] Error 1
make: Leaving directory `/home/vagrant/blah/node_modules/posix/build'

--- a/src/posix.cc
+++ b/src/posix.cc
@@ -650,11 +650,13 @@ NAN_METHOD(node_update_swap_constants) {
 
     Local<Object> obj = info[0]->ToObject();
     obj->Set(Nan::New<String>("prefer").ToLocalChecked(), Nan::New<Integer>(SWAP_FLAG_PREFER));
+#ifdef SWAP_FLAG_DISCARD
     obj->Set(Nan::New<String>("discard").ToLocalChecked(), Nan::New<Integer>(SWAP_FLAG_DISCARD));
+#endif // SWAP_FLAG_DISCARD
 
     info.GetReturnValue().Set(Nan::Undefined());
 }
-#endif
+#endif // __linux__
 
 #define EXPORT(name, symbol) exports->Set( \
   Nan::New<String>(name).ToLocalChecked(), \

